contracts/libraries/JBMetadataResolver.sol

```diff
@@ -51,7 +51,7 @@ library JBMetadataResolver {
      * @return _found          Whether the {id:data} was found
      * @return _targetData The data for the ID (can be empty)
      */
-    function getMetadata(bytes4 _id, bytes calldata _metadata)
+    function getHookMetadata(bytes4 _id, bytes calldata _metadata)
```

Comment:
Actually, to be in line with the rest of the lib, this should be `getData` (as we parse the metadata to find the data of a given id)

Why "hook"? This should now be usage agnostic (that's why I removed the delegate references in the non-language touched-up PR), no issue for this lib to be used in whatever metadata